libraries/amqp/include/eosio/amqp/amqp_handler.hpp

```diff
@@ -99,10 +99,18 @@ class amqp {
          std::promise<void> stop_promise;
          auto future = stop_promise.get_future();
          boost::asio::post( *handler_->amqp_strand(), [&]() {
-            if( channel_ ) {
+            if( channel_ && channel_->usable() ) {
                auto& cb = channel_->close();
                cb.onFinalize( [&]() {
-                  stop_promise.set_value();
+                  try {
+                     stop_promise.set_value();
+                  } catch(...) {}
```

Comment:
How would you get more that one call to `set_value`?  From [future.promise] it looks to me like that is the only way `set_value` will throw for `promise<void>`.

Reply:
I thought it might be possible for channel to call both onError and onFinalize.

Reply:
I see.